core/kazoo_tasks/src/kz_task_worker.erl

```diff
@@ -0,0 +1,163 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2016, 2600Hz INC
+%%% @doc
+%%% Run tasks scheduled by parent.
+%%% @end
+%%% @contributors
+%%%   Pierre Fenoll
+%%%-------------------------------------------------------------------
+-module(kz_task_worker).
+-behaviour(gen_server).
+
+%% API
+-export([start_link/4
+        ]).
+
+%% gen_server callbacks
+-export([init/1
+        ,handle_call/3
+        ,handle_cast/2
+        ,handle_info/2
+        ,terminate/2
+        ,code_change/3
+        ]).
+
+-include("kz_tasks.hrl").
+
+-define(SERVER, ?MODULE).
+
+-record(state, { task_pid :: pid()
+               }).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% @end
+%%--------------------------------------------------------------------
+-spec start_link(kz_tasks:task_id(), module(), atom(), list()) -> startlink_ret().
+start_link(TaskId, M, F, A) ->
+    _ = kz_util:put_callid(TaskId),
+    gen_server:start_link(?SERVER, [TaskId, M, F, A], []).
+
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Initializes the server
+%%
+%% @spec init(Args) -> {ok, State} |
+%%                     {ok, State, Timeout} |
+%%                     ignore |
+%%                     {stop, Reason}
+%% @end
+%%--------------------------------------------------------------------
+init([TaskId, M, F, A]) ->
+    Pid =
+        spawn_link(
+          fun () ->
+                  _ = kz_tasks:worker_running(TaskId, self()),
+                  try
+                      R = apply(M, F, A),
+                      _ = kz_tasks:worker_finished(TaskId),
+                      lager:debug("R = ~p. Should store this in MoDB", [R])
+                  catch
+                      _E:_R ->
+                          Stacktrace = erlang:get_stacktrace(),
+                          Msg = io_lib:format("(~p) ~p: ~w\n~p"
```

Comment:
That's what my fingers type automatically, also it's a standard escaped char instead of a special occurence from The Prolog Days™